### ISSUE TITLE
Packaging multiple models into one?

### ISSUE THREAD

**Issue by Robert-Forrest** [tensorflow/decision-forests#34]
Hi, thanks for this package!

I'm looking into converting an existing neural network model to use the decision forests approach, for comparison and/or use in an ensemble.

The existing neural network I've been developing has multiple outputs (mixture of regression and classification), and some of these outputs feed into other outputs (the other outputs also have access to the rest of the training data). 

It might be easier to explain via simple example (in the likely case that I'm not putting it into words well!). Let's say I have input features **In** and 3 target labels **A**, **B**, and **C**. My neural network works something like this:

**A** = Model(**In**)
**B** = Model(**In** + **A**)
**C** = Model(**In** + **A** + **B**)

This gives me a unified model for **A**, **B,** and **C,** which can be trained, saved, and loaded as one entity.

I can see how I might achieve something like this with decision-forests by using the preprocessing argument, passing in the training data and the previous model, and returning the training data with an added column for the prediction of the previous model. The final model would give a single output, but I could write something to load each model and make a list of all the predictions. In a similar vein, I could write something to load and augment the training data before the training of each model, as an alternative to using preprocessing.

Is there a way to obtain multi-output models in a way that is nicer than the (potentially silly) approach above?

Thanks for any help, sorry if this question doesn't make much sense, I can clarify if needed!

**Comment by achoum** [comment]
Hello Robert-Forrest :),

I think you listed correctly the two possible approaches: Compose the model during training (with preprocessing) or materialize explicitly the intermediate training dataset. I don't see a third one :).

This [other reply](https://discuss.tensorflow.org/t/add-tfdf-outputs-in-a-concatenate-api-functional-layer/1449) might be interesting for you.

Note that once the models are trained, you can pack them all into a single "super" model which can be use (serialization, prediction, evaluation) as a classical keras model.

M.
